### PR TITLE
feat: allow `--no-locked` for docker build, forbid for docker deploy; doc/guidance

### PR DESCRIPTION
**Pull request by dj8yfo**
photo_2024-07-05_13-28-09

### deployed after lock added

lock added with https://github.com/dj8yfo/near-sdk-rs/commit/ad04e7f11e546d751807fba58eb14b398625ff3f,
deployed to `test-adder-sdk.testnet` 
from https://github.com/dj8yfo/near-sdk-rs/tree/d4fb3166fd529554cd4e4cfec26bfe3f1505cf77/examples/adder

### CODE REVIEW

**Comment by dj8yfo** [comment]
@race-of-sloths, please include my PR in the race

**Comment by race-of-sloths** [comment]
@dj8yfo Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
New Sloth joined the Race! Welcome!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/dj8yfo)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/dj8yfo)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Comment by dj8yfo** [comment]
@race-of-sloths score 1

**Comment by race-of-sloths** [comment]
### ❌ Self-scoring is not permitted. 
Let's keep it fair! 👍

**Comment by frol** [comment]
@race-of-sloths score 1

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@frol, thank you for scoring this pull request in the Race of Sloths!

File: README.md [hunk 1]
```diff
@@ -61,6 +61,17 @@ $ cargo install --locked --path cargo-near
...
+
+If you want a less stressful experience, and just want to run a regular once-used-to-be
+build, run 
+
+```bash
+# equivalent to behaviour of `cargo near build` of version < 0.7.0 
+cargo near build --no-docker --no-locked
+``` 
+instead of `cargo near build`.
+
```
Comment (frol) [inline README.md:74]: This sounds like we discourage `cargo near build` and even moved it upfront. This is not the right message to deliver. Please, make it less aggressive and prominent

File: README.md [hunk 2]
```diff
@@ -93,10 +104,15 @@ By default, this runs a reproducible build in a [Docker](https://docs.docker.com
...
 - will use configuration in `[package.metadata.near.reproducible_build]` section of contract's `Cargo.toml`
 
-`--no-docker` flag can be used to perform a regular build with rust toolchain installed onto host, running the `cargo-near` cli. *NO*-Docker builds run against actual state of code in filesystem and not against a version, committed to source control.   
+`--no-docker` flag can be used to perform a regular build with rust toolchain installed onto host, running the `cargo-near` cli. 
+  - *NO*-Docker builds run against actual state of code in filesystem and not against a version, committed to source control.   
 
 `--no-locked` flag is allowed in *NO*-Docker builds, e.g. to generate a `Cargo.lock` *and* simultaneously build the contract.
 
+`--no-locked` flag is allowed in Docker builds, but 
+  - such builds are considered effectively DEMO-mode
```
Comment (frol) [inline README.md:113]: Before this feature, we had 4 years of NEAR living just fine with non-reproducible builds, let's not discourage it too much. Such contracts are still valid contracts and can hold millions $ in assets.

```suggestion
```

File: cargo-near/src/commands/build_command/docker.rs [hunk 3]
```diff
@@ -438,3 +462,23 @@ impl EnvVars {
...
+    };
+    println!(
+        "{}{}",
+        warning,
+        "Please mind that `--no-locked` flag is allowed in Docker builds, but:".cyan()
+    );
+    println!(
+        "{}",
+        "  - such builds are considered effectively DEMO-mode".yellow()
+    );
```
Comment (frol) [inline cargo-near/src/commands/build_command/docker.rs:481]:
```suggestion
```

**Review by frol — CHANGES_REQUESTED** [review]
Overall looks good to me, but please, address the review comments

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@frol, thank you for scoring this pull request in the Race of Sloths!

**Comment by race-of-sloths** [comment]
### ✅ PR is finalized! 
Your contribution is much appreciated with a final score of 5!
You have received 50 Sloth points for this contribution

**Pull request merged**